open-api/rest-catalog-open-api.yaml

```diff
@@ -2774,6 +3014,60 @@ components:
           additionalProperties:
             type: string
 
+    PlanTableResult:
+      type: object
+      description: If the plan has not finished return a `plan-id`. 
+        If finished, the response will contain a list of `FileScanTask`, a list of `PlanTask`, or both.
+      properties:
+        file-scan-tasks:
+          type: array
+          items:
+            $ref: '#/components/schemas/FileScanTask'
+        plan-tasks:
+          type: array
+          items:
+              $ref: '#/components/schemas/PlanTask'
+        plan-id:
+            description: id used to track progress of the plan
+            type: string
+
+    GetTasksStatusResult:
+      type: object
+      description: If the plan has not finished return a `plan-status`.
+        If the plan has finished can return a list of `FileScanTask`, a list of `PlanTask`, or both.
+      properties:
+        file-scan-tasks:
+          type: array
+          items:
+            $ref: '#/components/schemas/FileScanTask'
+        plan-tasks:
+          type: array
+          items:
+            $ref: '#/components/schemas/PlanTask'
+        plan-status:
+            $ref: '#/components/schemas/PlanStatus'
+
+
+    RetrieveTasksResult:
+      type: object
+      description: Used to fetch file scan tasks for a given `planTask`. Can also return additional plan-tasks.
+      properties:
+        file-scan-tasks:
+          type: array
+          items:
+            $ref: '#/components/schemas/FileScanTask'
+        plan-tasks:
+          type: array
+          items:
+            $ref: '#/components/schemas/PlanTask'
+
+    CancelPlanResult:
```

Comment:
We should drop this (see comment above on the delete route)

Reply:
will fix